app/src/main/java/org/p2p/wallet/history/signature/HistoryServiceSignatureFieldGenerator.kt

```diff
@@ -1,23 +1,26 @@
 package org.p2p.wallet.history.signature
 
+import kotlinx.coroutines.withContext
 import org.p2p.wallet.auth.gateway.repository.mapper.BorshSerializable
 import org.p2p.wallet.auth.gateway.repository.mapper.PushServiceSignatureFieldGenerator
 import org.p2p.wallet.auth.gateway.repository.mapper.write
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
 import org.p2p.wallet.utils.Base58String
 import org.p2p.wallet.utils.toBase58Instance
 import java.util.Optional
 
 class HistoryServiceSignatureFieldGenerator(
-    private val pushServiceSignatureFieldGenerator: PushServiceSignatureFieldGenerator
+    private val pushServiceSignatureFieldGenerator: PushServiceSignatureFieldGenerator,
+    private val coroutineDispatchers: CoroutineDispatchers
 ) {
-    fun generateSignature(
+    suspend fun generateSignature(
```

Comment:
generating signature is made in mappers, mappers are located and called in repositories that are already have withContext on site. 
there are a lot of withContext here and there, code became inconsistent